torch/testing/_internal/common_nn.py

```diff
@@ -5084,14 +5082,13 @@ def apply_fn(input, *params):
             def apply_fn(input1, input2, *params):
                 return module(input1, input2, target)
 
-        # TODO: we don't pass `target` as part of inputs because we don't
-        # currently compute the gradient w.r.t. target for loss functions.
-        gradcheck(apply_fn, inputs)
+        if target.requires_grad:
+            inputs = inputs + (target,)
```

Comment:
I think this works but it is a bit fragile.
Your `apply_fn` above actually ignores the target that is given in `*params`. But it works because it captures the same target from the parent scope. And the gradcheck ensures that you can do that. But I think it would be clearer to just get the target from the `apply_fn` inputs properly no?